src/commands/mod.rs

```diff
@@ -2490,24 +2490,34 @@ fn cmd_restore(
     args: &RestoreArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
+    let mut rebasing_from_parents = false; // The ways of using `Option` I can think of make this uglier.
```

Comment:
I don't quite follow the comment. Do you mean that the way we currently use `Option` makes this uglier than it has to be?

Either way, do you think we can instead extract the code from `cmd_diff()`?  That code currently calculates `from_tree` and a `to_tree`, but it could easily calculate a `from_tree` and a `to_commit` instead.

Reply:
Never mind the comment. The snippet from cmd_diff is much better.

Reply:
Done.